File: examples/hello_world/Android/shared/src/main/java/com/example/counter/shared_types/Requests.java
package com.example.counter.shared_types;

public final class Requests {

  public static java.util.List<Request> bincodeDeserialize(byte[] input) throws com.novi.serde.DeserializationError {
    if (input == null) {
      throw new com.novi.serde.DeserializationError("Cannot deserialize null array");
    }
    com.novi.serde.Deserializer deserializer = new com.novi.bincode.BincodeDeserializer(input);
    deserializer.increase_container_depth();

    long length = deserializer.deserialize_len();

    java.util.List<Request> value = new java.util.ArrayList<>();

    for (int i = 0; i < length; i++) {
      value.add(Request.deserialize(deserializer));
    }

    deserializer.decrease_container_depth();

    if (deserializer.get_buffer_offset() < input.length) {
      throw new com.novi.serde.DeserializationError("Some input bytes were not read");
    }
    return value;
  }
}

File: crux_core/typegen_extensions/csharp/Requests.cs
;

using System.Collections.Generic;
using Facet.Runtime.Bincode;
using Facet.Runtime.Serde;

public static class Requests
{
    public static List<Request> BincodeDeserialize(byte[] input)
    {
        if (input is null || input.Length == 0)
        {
            throw new DeserializationError("Cannot deserialize null or empty input");
        }

        var deserializer = new BincodeDeserializer(input);
        var length = deserializer.DeserializeLen();
        var value = new List<Request>((int)length);
        for (ulong i = 0; i < length; i++)
        {
            value.Add(Request.Deserialize(deserializer));
        }

        if (deserializer.GetBufferOffset() < input.Length)
        {
            throw new DeserializationError("Some input bytes were not read");
        }

        return value;
    }
}
